### PR TITLE
DOC: fix Sphinx Gallery discussion to explain mixed subddirs

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Update the "Writing documentation" for the changes from #25395 to point out the possibility of galleries with *.py and *.rst files in them (with links to the SG documentation about this). 

Also added a table of galleries with their sources and where they end up in the build.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by jklymak** [comment]
... draft for now - wording not right...

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -818,12 +816,28 @@ code will also appear in interactive docstrings.
...
+:file:`galleries/examples`       :file:`doc/gallery`
+:file:`galleries/tutorials`      :file:`doc/tutorials`
+:file:`galleries/users_explain`  :file:`doc/users/explain`
+===============================  ==========================
+
+The first three are traditional galleries, :file:`galleries/users_explain` is a
+mixed gallery where some of the files are raw ``*.rst`` files rather than
+``*.py`` files; Sphinx Gallery just copies those from the location to the build
+location (see :ref:`raw_restructured_gallery`, below). To exclude an example
 from having an plot generated insert "sgskip" somewhere in the filename.
```
Comment (melissawm) [inline doc/devel/documenting_mpl.rst:839]: There may be a sentence part missing here?

**Review by melissawm — COMMENTED** [review]
This looks great to me!

File: doc/devel/documenting_mpl.rst (hunk 1 above)
Comment (jklymak) [reply doc/devel/documenting_mpl.rst:839]: Maybe it's correct English, but I agree it's not very easy to parse.  I'll take another swing at it.

Comment (jklymak) [reply doc/devel/documenting_mpl.rst]: OK, modified to be a little more clear (I hope)

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -20,12 +20,10 @@ the docstrings of the classes in the Matplotlib library.  Except for
...
 when the documentation is built.  See :ref:`writing-docstrings` below.
 
-Second, the contents of :file:`doc/plot_types`, :file:`doc/gallery` and
-:file:`doc/tutorials` are generated by the `Sphinx Gallery`_ from python files
-in :file:`galleries/plot_types/`, :file:`galleries/examples/` and
-:file:`galleries/tutorials/`. These sources consist of python scripts that have
-ReST_ documentation built into their comments.  See
-:ref:`writing-examples-and-tutorials` below.
+Second, our example pages, tutorials, and some of the narrative documentation
+are created by `Sphinx Gallery`_.  Sphinx Gallery converts example python files
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:24]:
```suggestion
are created by `Sphinx Gallery`_.  Sphinx Gallery converts example Python files
```

File: doc/devel/documenting_mpl.rst [hunk 3]
```diff
@@ -818,11 +816,30 @@ code will also appear in interactive docstrings.
 Writing examples and tutorials
 ==============================
 
-Examples and tutorials are python scripts that are run by `Sphinx Gallery`_
-to create a gallery of images in the :file:`/doc/gallery` and
-:file:`/doc/tutorials` directories respectively.  To exclude an example
-from having an plot generated insert "sgskip" somewhere in the filename.
-
+Examples and tutorials are python scripts that are run by `Sphinx Gallery`_.
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:819]:
```suggestion
Examples and tutorials are Python scripts that are run by `Sphinx Gallery`_.
```

File: doc/devel/documenting_mpl.rst [hunk 4]
```diff
@@ -818,11 +816,30 @@ code will also appear in interactive docstrings.
...
+:file:`galleries/users_explain`  :file:`doc/users/explain`
+===============================  ==========================
+
+The first three are traditional galleries.  The last,
+:file:`galleries/users_explain`, is a mixed gallery where some of the files are
+raw ``*.rst`` files and some are ``*.py`` files; Sphinx Gallery just copies
+these ``*.rst`` files from the source location to the build location (see
+:ref:`raw_restructured_gallery`, below).
+
+In the python files, to exclude an example from having a plot generated, insert
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:841]:
```suggestion
In the Python files, to exclude an example from having a plot generated, insert
```

File: doc/devel/documenting_mpl.rst [hunk 5]
```diff
@@ -947,6 +964,23 @@ to name your example consistently, i.e. use the main function or subject
 of the example as first word in the filename; e.g. an image example
 should ideally be named similar to :file:`imshow_mynewexample.py`.
 
+.. _raw_restructured_gallery:
+
+Raw restructured text files in the gallery
+------------------------------------------
+
+`Sphinx Gallery`_ folders usually consist of a ``README.txt`` and a series of
+python source files that are then translated to an ``index.rst`` file and a
```
Comment (QuLogic) [inline doc/devel/documenting_mpl.rst:973]:
```suggestion
Python source files that are then translated to an ``index.rst`` file and a
```

**Review by QuLogic — APPROVED** [review]
This will need a rebase to fix CircleCI.

**Pull request merged**